src/Middlewares/Handler/Producer.php

```diff
@@ -1,89 +1,45 @@
 <?php
 namespace Metamorphosis\Middlewares\Handler;
 
-use Metamorphosis\Connectors\Producer\Connector;
-use Metamorphosis\Facades\Manager;
 use Metamorphosis\Middlewares\MiddlewareInterface;
+use Metamorphosis\Producer\Poll;
 use Metamorphosis\Record\RecordInterface;
-use Metamorphosis\TopicHandler\Producer\HandlerInterface;
-use RuntimeException;
+use RdKafka\ProducerTopic;
 
 class Producer implements MiddlewareInterface
 {
     /**
-     * @var Connector
-     */
-    private $connector;
-
-    /**
-     * @var HandlerInterface
-     */
-    private $producerHandler;
-
-    /**
-     * @var \RdKafka\Producer
+     * @var int
      */
-    private $producer;
+    private $partition;
 
     /**
-     * @var int
+     * @var \RdKafka\ProducerTopic
      */
-    private $processMessageCount = 0;
+    private $topic;
 
     /**
-     * @var int
+     * @var Poll
      */
-    private $partition;
+    private $poll;
 
-    public function __construct(Connector $connector, HandlerInterface $producerHandler)
+    public function __construct(ProducerTopic $topic, Poll $poll, int $partition)
     {
-        $this->connector = $connector;
-        $this->producerHandler = $producerHandler;
-
-        $this->producer = $this->connector->getProducerTopic($producerHandler);
-        $this->topic = $this->producer->newTopic(Manager::get('topic_id'));
-        $this->partition = Manager::get('partition');
+        $this->topic = $topic;
+        $this->poll = $poll;
+        $this->partition = $partition;
     }
 
     public function process(RecordInterface $record, MiddlewareHandlerInterface $handler): void
     {
         $this->topic->produce($this->getPartition($record), 0, $record->getPayload(), $record->getKey());
-        $this->handleResponse();
-    }
-
-    public function __destruct()
-    {
-        $this->flushMessage();
-    }
-
-    private function handleResponse(): void
-    {
-        $this->processMessageCount++;
-
-        if (!Manager::get('is_async')) {
-            $this->flushMessage();
-
-            return;
-        }
 
-        if (0 === ($this->processMessageCount % Manager::get('max_poll_records'))) {
-            $this->flushMessage();
-        }
+        $this->poll->handleResponse();
     }
 
-    private function flushMessage(): void
+    public function __destruct()
     {
-        if (!Manager::get('required_acknowledgment')) {
-            return;
-        }
-
-        for ($flushAttempts = 0; $flushAttempts < Manager::get('flush_attempts'); $flushAttempts++) {
-            if (0 === $this->producer->poll(Manager::get('timeout'))) {
-                return;
-            }
-        }
-
-        throw new RuntimeException('Unable to flush, messages might be lost!');
+        $this->poll->flushMessage();
```

Comment:
I was thinking about it now, we should use `register_shutdown_function()` instead of `__destruct()`, because it doesn't get call on fatal errors.
we could change this in the future, if this makes sense.